### ISSUE TITLE
Allow update assignment on multiple values

### ISSUE THREAD

**Issue by s-ol** [leafo/moonscript#373]
e.g.

```moon
a, b = 1, 2
a, b += 3, 4
print a, b # 4, 6
```

I think this would be very handy, especially working with tuples like 'light' vectors (x,y).

maybe even
```moon
a, b = 1, 2
a, b += 3
print a, b # 4 5
```
though I am not sure how useful and idomatic that would be.

**Comment by RyanSquared** [comment]
> maybe even
>```moon
>a, b = 1, 2
>a, b += 3
>print a, b # 4 5
>```

absolutely not, because:

```moon
x = ()-> 1, 2
a, b += (x!)
a, b += x!
```

You'd have to be aware every time of the vararg list, which can cause confusions. I tend to have similar issues with the `string` library and would hate to see the opposite issue with update assignment.

**Comment by s-ol** [comment]
@RyanSquared good catch on the multi-returns, I felt like there was something wrong with that syntax but didn't figure it out.

Your elaborated Lua is the behaviour that you **don't** want to see, right?

**Comment by RyanSquared** [comment]
I'd like to not have the syntax at all because I'd prefer for MoonScript to be as little overhead as possible; this would involve an overhead which might not be necessary.

Additionally, I'm not fond of the syntax at all as it tends to lead to confusion.

**Comment by s-ol** [comment]
> I'd like to not have the syntax at all because I'd prefer for MoonScript to be as little overhead as possible; this would involve an overhead which might not be necessary.
> 
> Additionally, I'm not fond of the syntax at all as it tends to lead to confusion.

well i think the syntax should not involve any checking and be plain sugar with a matched amount of names / values on both sides.

**Comment by RyanSquared** [comment]
The issue is that it is literally impossible to have a matched amount of values on the RHS unless you completely get rid of vararg returns. Alternatively you can assert() return values from a vararg return but that also has overhead.

**Comment by s-ol** [comment]
> The issue is that it is literally impossible to have a matched amount of values on the RHS unless you completely get rid of vararg returns. Alternatively you can assert() return values from a vararg return but that also has overhead.

why? you just transform

    a, b, c, d [OP]= [ANYTHING]

into

    local _tmp_a, _tmp_b, _tmp_c, _tmp_d = [ANYTHING]
    a, b, c, d = a OP _tmp_a, b OP _tmp_b, c OP _tmp_c, d OP _tmp_d

if you make a mistake, there will be an error thrown (operation on nil value) at runtime, as there should be, since this is a runtime problem occuring.
